### PR TITLE
WRD Homepage Takeover Driven by contentful setting

### PR DESCRIPTION
**Pull request by eddieferrer**
* Uses a contentful setting to drive homepage content

LW-1956

### CODE REVIEW

**Comment by eddieferrer** [comment]
In my dev vm I get: 
vendors~app.js:99356 [Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside <p>, or missing <tbody>. Bailing hydration and performing full client-side render.

I burned some time trying to fix this issue and got nowhere. Everything seems to be working, the issue seems to point to the kvResponsiveImage component. I tried setting cache keys on it, and modifying the selection of the homepage component in homepage.vue several ways, and nothing would make it go away, any suggestions there are appreciated. 

I dont think this warning is an issue though, and the homepage takeover will still work.

File: src/pages/Homepage/DefaultHomepage.vue [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <template>
 	<div>
-		<top-message-contentful />
```
Comment (eddieferrer) [inline src/pages/Homepage/DefaultHomepage.vue:3]: I don't believe this should be here. This component doesnt even render without content prop

File: src/pages/Homepage/Homepage.vue [hunk 2]
```diff
@@ -72,26 +69,33 @@ export default {
...
+			const contentSetting = _get(data, 'contentful.entries.items', []).find(item => item.fields.key === 'ui-homepage-takeover'); // eslint-disable-line max-len
+			if (_get(contentSetting, 'fields')) {
 				const isContentEnabled = settingEnabled(
 					contentSetting.fields,
 					'active',
 					'startDate',
 					'endDate'
 				);
 				if (isContentEnabled) {
+					const takeOverKey = _get(contentSetting, 'fields.dataObject.takeOverKey');
```
Comment (eddieferrer) [inline src/pages/Homepage/Homepage.vue:82]: I think this is a cleaner way to pick 1 homepage takeover design based on a single setting in contentful

File: src/pages/Homepage/iwd/IWDHomepage.vue [hunk 3]
```diff
@@ -342,7 +341,6 @@ export default {
 		NumberCountries,
 		WavyDivider,
 	},
-	inject: ['apollo'],
```
Comment (eddieferrer) [inline src/pages/Homepage/iwd/IWDHomepage.vue:345]: Not needed

File: src/pages/Homepage/iwd/IWDHomepage.vue [hunk 4]
```diff
@@ -11,7 +11,6 @@
 		>
 			<kv-responsive-image
 				:images="heroImageSet"
-				alt=""
```
Comment (eddieferrer) [inline src/pages/Homepage/iwd/IWDHomepage.vue:14]: Not needed, kv responsive wont even pass attributes down.

**Comment by mcstover** [comment]
@eddieferrer Turning of JS reveals an interesting comparison of rendered html. Without JS, the clickable `.hero` anchor tag is a sibling to the `.hero__text-wrapper` and `svg`... However, the Vue rendered html places `.hero__text-wrapper` and `svg` within the anchor...

The mismatch still exists even with the responsive image component removed...

Certainly not ideal but also not a blocker for such a tight timeline...

**Comment by mcstover** [comment]
👍 +1

File: src/pages/Homepage/iwd/IWDHomepage.vue (hunk 4 above)
Comment (ryan-ludwig) [reply src/pages/Homepage/iwd/IWDHomepage.vue:14]: @mcstover @eddieferrer A small thing, but we should always have some alt tag on images. `alt=""` tells a screenreader that this is a decorative image and to ignore it. Having no alt attribute [makes some screenreaders announce the image URL](https://www.w3.org/WAI/tutorials/images/decorative/). 

kv-responsive-image v-binds $attrs to the `img` inside. We could add a non-required alt prop, and if there's no prop set alt to "" on the img inside. Though I kindof like reinforcing the always alt pattern so it's remembered for regular images.

Comment (mcstover) [reply src/pages/Homepage/iwd/IWDHomepage.vue:14]: This is a good callout. I think we just got caught up in the rush.

Comment (eddieferrer) [reply src/pages/Homepage/iwd/IWDHomepage.vue:14]: @ryan good point, I was incorrect about my comment above. Ill revert this

**Pull request merged**